### ISSUE TITLE
Avoid computing computed with `storeToRefs()`

### ISSUE THREAD

**Issue by Ericlm** [posva/pinia#2812]
### Reproduction

https://github.com/Ericlm/pinia-computed

### Steps to reproduce the bug

Since version 2.2.5, computed values are evaluated at the start even if they're not displayed (when using storeToRefs).

It may be related to https://github.com/vuejs/pinia/commit/254eec764ecccd120ab9dbc31352d76ffd6ecfa3

### Expected behavior

The computed should only be evaluated when it is actually used.

### Actual behavior

The computed is evaluated even if its value is displayed inside a `v-if="false"`.

**Comment by posva** [comment]
This seems to be inherent to Vue's `toRef()`. The same behavior [exists in regular Vue](https://play.vuejs.org/#eNp9Uk1v2zAM/SuELnEHwx22nTonwFb0sB22oeut6sG1GcepLAkSlQYw/N9LSUmbAEVvIt974uPHJH5YW+0CiitR+9YNlsAjBQuq0f1SCvJSrKQeRmscwQQOm5aGHcIMa2dGWLB28f2MsC6hNaMNhF0JZG5x7Ut4bqjdHMIzrdRSAwu0J1Cm97CMX9Se3KD7+4dVcf9w8cZoTdCUKcXnk3xnwqPC6wN6LF8UF7BcwRR5kH6vdo0KWNngN8XiVJQkSN2CP810x2NwOlfMMvgEXyI2Hzi5tHncJkN5LsWUFeWZpaiAwywKFnBYX+Zx83A5IBytagg5Aqitw9U0Rb/zXF/GKPJPOKLkxXD59dBXW280by81KUXsY1Do/loa2J4UV8f2pWiUMs+/U45cwPKYbzfYPr2T3/p9zEnxz6FHt0MpXjFqXI+U4Zv/f3DP71dwNF1QzP4AvEVvVIgeM+1n0B3bPuElt7/SWfEl3PmbPaH2x6ai0bSLxJeCT+n6g9bf7H6tviWd1LOYXwAIIAIt). If you want to avoid computation, you can directly reference the store with `store.doubleCount` or pass `() => store.doubleCount`.

A possible improvement here would be to change the current implementation related to HMR in order to keep using the old version in `storeToRefs()` that references the raw store object (thus not triggering reactivity). Contribution welcome.

Edit: I'm not completely sure we want to revert this change because if this behavior exists in current Vue, it's more natural to keep it. That being said I would like to have real world examples of why this change affects you

**Comment by sceee** [comment]
I understand that this might be according to Vue's `toRef()`, but nevertheless this seems to be like a pretty big change in behavior.
In my case, it seems to "break" a lot of stuff as it just behaved different until now and therefore was used in that way.

@posva as I'm not sure I understood your tips regarding avoiding computation correctly (and I can unfortunately also not view your "play" link as it throws an exception `InvalidCharacterError: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.` which seems to be a general vue play error at the moment):

What exactly do you mean with "you can directly reference the store"?
To use this
```typescript
const { myComputedProp } = myStore
```
instead of
```typescript
const { myComputedProp } = storeToRefs(myStore)
```
?
But this is then no longer reactive, is it?

Or what do you mean with your other mentioned fix "pass () => `store.doubleComputed`"?
This:
```typescript
const myComputedProp = computed(() => myStore.myComputedProp)
```
?

Thanks.

**Comment by posva** [comment]
I updated my comment with a fixed link. In your case, it seems like you can simply not use `storeToRefs()` at all. The real use case for it is to pass the extracted ref to another composable. Most of the time, these composables also accept a getter function `() => store.computed`, which is why I'm suggesting the getter

> In my case, it seems to "break" a lot of stuff as it just behaved different until now and therefore was used in that way.

Interested in learning more and [available to help as well!](https://cal.com/posva/consultancy?date=2024-10-31&month=2024-10)

**Comment by marr** [comment]
> Most of the time, these composables also accept a getter function () => store.computed, which is why I'm suggesting the getter

I'd like to understand this more thoroughly. In the app I'm working on, we seem to use `storeToRefs` all over the place. I use computed quite often and thought I would have to use the `.value` of the transformed (via storeToRefs) ref. If this is not the case, can you tell us how that works? Since `computed` depends on reactive values, how would it work just referencing `store.foo` since that doesn't seem to be reactive?

**Comment by posva** [comment]
store.foo is reactive because store is a reactive object

**Comment by marr** [comment]
If I do something like this, I don't see updates:

```
  watchEffect(() => {
    console.log(store.foo)
  });
```

**Comment by marr** [comment]
Is there something wrong with how I'm using the store here?
https://stackblitz.com/edit/nuxt-pinia-ssr-qwpymm?file=app.vue

**Comment by posva** [comment]
Yes, you are returning a plain string:

![CleanShot 2024-10-31 at 10 18 31](https://github.com/user-attachments/assets/54f2ae4a-2d9d-40bc-80c4-0ae2984f8d3c)

But it seems that the `t` function doesn't keep reactivity because it doesn't work within a computed either. If I were you, I would check with nuxt i18n

Otherwise, [it works](https://play.pinia.vuejs.org/#eNqNVE1vGjEQ/SvWXiAN2EloUymFiLbKIT20VdPjXpbdgTjxl+wxQUL8987aQDakieC09syb92Y8vHXx1Tm+jFBcFeNQe+mQBcDomKrMYlIWGMriujRSO+uRrVkMcIfWA9uwubea9biobTQInmPodRKfKqzvb+ZzqHGfSzyUUpramkA8qc5kX7J/0sa6wH7/hE2u2Zpu6deirAKu7KKfsLyxcaaAYIxtTr6UZixyCySYDgjaqQqBToyNZxHRGjatlawfqbFcwZyeUn+3pvagwZDudZbFDdtsxiKDqMBYdKoVgyK3OdSV4w/BGppeElluAzS0q53ssqC223NZ3CO6cCVE3RiCNaDk0nMDKIzTYkppwtMopYZhY/V0xD/x8wvRyIDdew5BD2fePgWa+QMxDTo8gi6X4IceTAMe/LG8B7AX3Aext/mdNLLKjCJ9v52aSBtYorUqDCsnjxXaxUwv+SU/F0rOBBEJSfpWryYybEAfU32XOj3j5x/551R1OdoW1W3VtGal2dDzY6BlnMvFwePXVjupwP9yKGlZXyxBpZR9+pHu0EfYS6zvoX78z/1DWGXVvz2kF+i0hZVfAObwzd1PWNH3PqhtE9V24d4I/gH6I8VWY077Fk1Dsjt5Se1t2mVpFn/DzQrBhF1TrdA0jZSfxvz9ndaf5Y74qDPFZ+OgEe6No4G5NAcmk7ap6y4e5gMyBO0iQvPKX2CV0rLN7B1r0i3d723ZewPWcZkMMZRLDP2zZC35LnsNBXasW3MyfFmpCOwDu8j+RUiM3pBIM9ih2oYpWmz+Acq/1w0=)

**Comment by chuchiperriman** [comment]
Same problem here. Variables are being evaluated before being read when we use storeToRef. 

This change is not compatible with previous functionality and breaks all components that use the store.

**Comment by alt1o** [comment]
Same problem here. It should be a break change.

**Comment by DaleMckeown** [comment]
I opened https://github.com/vuejs/pinia/issues/2829 earlier but it was closed down, which is fine as I was still struggling with how to replicate the problem in the play link.

I've spent 2 days debugging this issue in Pinia, and I finally understand what was happening in my project.

One store was breaking my app, with a completely white screen and lots of Vue warn errors:
```
 Slot "default" invoked outside of the render function: this will not track dependencies used in the slot. Invoke the slot function inside the render function instead. 
```

This specific store had some computed properties that were referencing other computed properties, like so:

```
export const useDataStore = defineStore('data', () => {

  const loadingData = ref(true);
  const data = ref<User[]>();

  async function loadData() {
    try {
      loadingData.value = true;

      setTimeout(() => {
       data.value = [
          { id: 1, name: 'Alice', userType: 'student', age: 20 },
          { id: 2, name: 'Bob', userType: 'student', age: 22  },
          { id: 3, name: 'Charlie', userType: 'teacher', age: 40  },
          { id: 4, name: 'Tina', userType: 'student', age: 21  },
          { id: 5, name: 'Steven', userType: 'teacher', age: 45  },
          { id: 6, name: 'Abdul', userType: 'student', age: 19  },
          { id: 7, name: 'Yet', userType: 'teacher', age: 50  }
        ];
      }, 2000);
    }
    catch (error) {
      console.log(error);
    }
    finally {
      loadingData.value = false;
    }
  }

  const allStudents = computed(() => {
    return data.value.filter(user => {
      return user.userType === 'student';
    });
  });

  const studentsUnder20 = computed(() => {
    return allStudents.value.filter(user => {
      return user.age < 20;
    });
  });

  return { loadingData, data, allStudents, studentsUnder20, loadData }

});
```

[Pinia Playground Link](https://play.pinia.vuejs.org/#eNqVVk2T2jgQ/SsqLjC1jD3fqZ3A1CTZHHYPu1sZctgKczB2A8raskuSCVMU/z3d+rIgwCQcwLT6tV4/dbe86b1rmmTVQu++N1K55I1mCnTbsDITi/G0p9W09zAVvGpqqdmHGn8FCM3msq5YP0mDhYL0307FKLVhEIR/NFRNmWnAf4yNCr4yD/gYcA+jLob1Sq3bKI3AvWHPUjivsib5qmqBhDfkPnULyPOeGQvZGi54RpZpLzXPCajqfCbrbwok4qe9oXd9ROJpAStd16U6zxpuYUutG3Wfpnkh0L+Akq9kIkCnoqnSHzCPd8ldcpmWfJbiRikXBax3t0HEeQHVz0T3ro8XyeVN8sZEXV27oBVFpaDbqdiiLFrltZjzxZ4oOWrKS5D/NJrXYlecrCzrb38Zm5YtBIr5EvL/D9i/qrVl/a8ElG8FUVo6kwvQdvnj09+wxuewWNVFW6L3icVPoOqyJY7W7X0rCqQd+Rm2f5oz5mIxUR/XGoTySRFRo4bxNzJTOR1LvaN7nVxHKhaZzp50LSHRCkV05b5hBcy5ALPCtq7oTT31Q09sGGndaiiGTMI8uFE/UA/olwbYZ9SNjT0PXtwz0VYzkC5LkVVwz5SWmKEztYiYIHTPnC3QYrHInOLD2tDAKlCaUH/4VHC/iP6gTzn2h2xwxsYPxIQCWlRZZwVuQUgEYRIDkvUMu9l7ENYujSiVL88PA1omh0y9iJzNW5GT3CYWBcJtXLZavvhHFm+VrLKyJZa0mQtGH5w/E15B3epB4GpXDI0A+xLMeAYk6eXQCdl/V/IcMNdOxL7SbYEjBo1GwqsLVzJxgKsQ4H09Ow2/8iUX468D/sMykyXfo6AhwxaTPsbNxaEYNyHGhAs6rxMkLg8FuA0BnjSsQJzmcHsoxF0n5Ayb9CSJy98PRXgTIvwH5HqcwS2p4OHPpuTosx3iEV1c2BoMHnmm8yUbgJS1DPVla7QuISnrhVvbhWET4BR5pQrnWamoDAPOdJdvAMQ/2cQV+vqO3y1QiRenFFGRJnNeapADyn6nkJ0n2RMvDRuPx526nojNxPx0ZJyT+ow3gsRKfoVQxP0XeOHpsBEewhEmznUTazk0uePBdhsO98kOw4SwAruAOIN3XiWi95H4TYTR1nvTOeCGbLkzfQ3h4KxoCk7qTzBXe6N8129nhDpHetMJVwT5G03oy01fZJK1pfmNGQ2csn4qeHvfNYx51bKTMq5lO2/9EI8JmakbO76qf9eZ+1WzxdiRKIOwpz9iP3HtxdgN9rfmLg0uvhK6nWJKndVw6/7+FMkwDozYdtL42iaP7pYJPJZ40/EVjpfofkBbgxa95Mrk4cidRRs7lz7SMvWvWJ/9ZhGmoUsQC708hvC0O1DcdMexAYevPdiLeCohwJ4SPwR5DkPOmEwfhZfv3vY7W4fC7Q==)

In 2.2.4 and below, the above was perfectly fine. In 2.2.5 and above, my computed properties **have** to use optional chaining, otherwise the store breaks due to that early evaluation:

```
const allStudents = computed(() => {
  return data.value?.filter(user => {
    return user.userType === 'student';
  });
});

const studentsUnder20 = computed(() => {
  return allStudents.value?.filter(user => {
    return user.age < 20;
  });
});
```

I also feel like the behaviour should not have been changed without warning. At the very least, docs should be updated to reflect the new behaviour.

**Comment by skirtles-code** [comment]
I just encountered someone having this problem on Vue Land.

Here's a reproduction of the problem they were having:

- [Playground](https://play.vuejs.org/#eNp9U01P4zAQ/Suj7KHpijpAC4eKou4H0u4edhGgPUVamWZaDI5t2U6pVPW/79gmbQqUQ9V45s3ze2+SdVZzodijy8aZqI22HtYws8g9fjEGNjC3uobessFeqV4BroUSfAsx4bQDhenUYAU3hiWGUm25c/r1SwXAGod5hzHvp3KtG+Xz3iea7vWzo6zing+c1xaZ35Nb4VwovA2dN2pKhasIm2nlPNBV34kmYSfdybxX64rL3hGsw+3Ok5wx5H2YXEIeSwBBwRhUI2U4b/pH4W+B3qN14zQHYKw2f7lsMI8c/bYOYNE3ViVqFrhYwMJnGCbEJrKWahPcvmRGPi/czArjwdG4AcnVYlJm3pXZZWcjMZc7fYNzd2gj6333LYoVe8GmJcWwQr1Nqjua035azHrnlxgnXR35dp7wF0VyQZrp4LE2klKgE8APlFIHRKdKASTVg5obeju1oiBikOVLg/xvIy+zKUVVVLj0Wks34EaEbpk9eG/cuCgaZZ4WbKbr4g1wes7O2aiQ4r5AVxdCVbiiC8ssLjdwxxgP8cXm9JSdsrOiEs6nAiOqwb3Vzw7tPhldP6iwPqiv7U+P2cmInRxHYUlUHYjaV4QC8o5WMBeLV/EQixES7R/jBa1oLyZOST//ijVvG9yqmj3g7Omd+qNbJaHXFsnKEjtOPLf06qf21e1vXNHztknfUiMJ/UHzBp2WTdCYYF8bVZHsDi6q/Rm3LdTizl2tPCrXmgpCYxoRH5P99oH1ndwhG3VS/Lekj5fQFOCQnbGT02zzH8UWtC0=)

They were using `propValue` conditionally in the template (inside a `v-if`), but now `storeToRefs` throws even without accessing `propValue.value`.

This works fine with 2.2.4 and earlier, but fails with 2.2.5. As noted by others, this seems to have been caused by #2795.

When `storeToRefs` calls `toRef` internally it eventually gets to this code in Vue:

```ts
function propertyToRef(
  source: Record<string, any>,
  key: string,
  defaultValue?: unknown,
) {
  const val = source[key]
  return isRef(val)
    ? val
    : (new ObjectRefImpl(source, key, defaultValue) as any)
}
```

In Pinia 2.2.4 and earlier the `source` object is the raw, unwrapped store object, so accessing `source[val]` is just grabbing the `ComputedRefImpl` for the getter out of that object. All works well.

With 2.2.5 the source object is now a proxy, so accessing `source[key]` tries to unwrap the `ComputedRefImpl`, causing the error.

Would the fix in Vue core be something like this?

```js
function propertyToRef(
  source: Record<string, any>,
  key: string,
  defaultValue?: unknown,
) {
  if (!isProxy(source)) {
    const val = source[key]
    if (isRef(val)) {
      return val
    }
  }
  return new ObjectRefImpl(source, key, defaultValue) as any
}
```

If `source` is a proxy (i.e. reactive or readonly) then it doesn't seem necessary to do the `isRef` check, as proxies will always unwrap them anyway.

*Update:* The code I suggested above isn't quite right, as it won't handle `shallowReactive`/`shallowReadonly` correctly. Handling those correctly is a bit fiddly, but I'm working on it.

**Comment by posva** [comment]
@DaleMckeown In your case, the fix is very simple: `const data = ref<User[]>([])` pass that initial value. You will notice that [you get a TS error](https://play.pinia.vuejs.org/#eNqVVk1z2zYQ/SsYXiRPZdLfmTqSx0maQ3toO7Fz6EQ+0ORKQkqCHABU5NHov2cXX4QUSU50kKjFvsXbh90F18m7tk2XHSS3yVgVkreaKdBdy6pczCfTRKtpcjcVvG4bqdmHBn8FCM1msqnZIM2ChYIM3k7FOLNhEIR/NNRtlWvAf4yNS740D/gYcHfjPob1yqzbOIvAySixFE7rvE2/qkYg4TW5T90C8rxlxkK2lguek2WaZOY5BVWfPsvmmwKJ+Gky8q73SDwrYambplKnecstbKF1q26zrCgF+pdQ8aVMBehMtHX2A+b+Jr1Jz7OKP2e4UcZFCavtbRBxWkL9M9G96/1Zen6VvjFRl5cuaE1RKehmKjYoi1ZFI2Z8viNKgZryCuQ/reaN2BZHackLTSYtOwgM86pqvv1lXLfsxQKK//fYv6qVTeZfCajqEqJsdS7nYLaYJh8f/oYVPofFuim7Cr2PLH4C1VQdUbdu7ztRYjaRn2H7pzl6LuaP6uNKg1A+VyJqRDL+Rn2qskOK9HQv08tI3DLX+YNuJKRaobauC9ashBkXYFbYxvWCKbNBaJU1oyPoNJQjJmEW3KhNqDX0SwvsM+rGJp4HL2+Z6OpnkC5Lkddwy+i4xNyZOkQ8InTHnM/RYrHInOLDytDA4lCaUH/4VHC/iP5wQDkORmx4wiZ3xIQCWlTV5CVuQUgEYRJDkvUEm9x7ENYujSmVL093Q1omh1y9iILNOlGQ3CYWBcJtXLZavvhHFm+VLvOqI5a0mQtGHxxLj7yGptPDwNWuGBoB9iWY8QxI0vORE3LwruIFYK69iAOluxInDxqNhBdnrmTiABchwPvm+Tj8wpdcjL8M+A+LXFZ8h4KGHFtM+hhXZ/tiXIUYj1zQeR0hcb4vwHUI8KBhCeI4h+t9IW56IZ+xSY+SOP99X4Q3IcJ/QK6HGVyTCh7+ZEqOPpsRHtHZma3B4FHkuliwIUjZyFBftkabCtKqmbu1bRg2AU6RV6pwlleKyjDgTHf5BkD8g01coa/v+O0ClXifShEVaTrjlQY5pOy3Ctl5kj310rDJZNKr64nYTMxPT8Y5qc94UUis5FcIRdx/gReeDhvjIRxg4lzXsZYjkzsebL/haJfsKEwIK7ALiDN46w0jek2JX1AYbb0znQNuxBZb09cQDs6KpuBj8wlmameUb/ttjVDnSC9A4Yogf6MJfbnpi0zyrjK/MaOhU9ZPBW8fuIYxb2B2Usa1bOetH+IxITN1Y8dX9e87c7dqNhg7EmUY9vRH7CeuvRj7wf7W3KXBxVdCv1NMqbcabv3fnyIZxoER204aX9vk0d8ygccCbzq+xPES3Q9oa9GiF1yZPBy5k2hj5zJAWqb+FRuw3yzCNHQFYq4XhxCedg+Km+4wNuDwtQd7EU8lBNhR4ocgT2HIGZPpo/BOnmy+A6dEyhI=)

**Comment by posva** [comment]
@skirtles-code I marked this upstream but I think we could and should handle getters and computed differently in `storeToRefs()` so they are still lazily evaluated. Keeping the HMR working is also needed though.

**Comment by DaleMckeown** [comment]
> @DaleMckeown In your case, the fix is very simple: `const data = ref<User[]>([])` pass that initial value. You will notice that [you get a TS error](https://play.pinia.vuejs.org/#eNqVVk1z2zYQ/SsYXiRPZdLfmTqSx0maQ3toO7Fz6EQ+0ORKQkqCHABU5NHov2cXX4QUSU50kKjFvsXbh90F18m7tk2XHSS3yVgVkreaKdBdy6pczCfTRKtpcjcVvG4bqdmHBn8FCM1msqnZIM2ChYIM3k7FOLNhEIR/NNRtlWvAf4yNS740D/gYcHfjPob1yqzbOIvAySixFE7rvE2/qkYg4TW5T90C8rxlxkK2lguek2WaZOY5BVWfPsvmmwKJ+Gky8q73SDwrYambplKnecstbKF1q26zrCgF+pdQ8aVMBehMtHX2A+b+Jr1Jz7OKP2e4UcZFCavtbRBxWkL9M9G96/1Zen6VvjFRl5cuaE1RKehmKjYoi1ZFI2Z8viNKgZryCuQ/reaN2BZHackLTSYtOwgM86pqvv1lXLfsxQKK//fYv6qVTeZfCajqEqJsdS7nYLaYJh8f/oYVPofFuim7Cr2PLH4C1VQdUbdu7ztRYjaRn2H7pzl6LuaP6uNKg1A+VyJqRDL+Rn2qskOK9HQv08tI3DLX+YNuJKRaobauC9ashBkXYFbYxvWCKbNBaJU1oyPoNJQjJmEW3KhNqDX0SwvsM+rGJp4HL2+Z6OpnkC5Lkddwy+i4xNyZOkQ8InTHnM/RYrHInOLDytDA4lCaUH/4VHC/iP5wQDkORmx4wiZ3xIQCWlTV5CVuQUgEYRJDkvUEm9x7ENYujSmVL093Q1omh1y9iILNOlGQ3CYWBcJtXLZavvhHFm+VLvOqI5a0mQtGHxxLj7yGptPDwNWuGBoB9iWY8QxI0vORE3LwruIFYK69iAOluxInDxqNhBdnrmTiABchwPvm+Tj8wpdcjL8M+A+LXFZ8h4KGHFtM+hhXZ/tiXIUYj1zQeR0hcb4vwHUI8KBhCeI4h+t9IW56IZ+xSY+SOP99X4Q3IcJ/QK6HGVyTCh7+ZEqOPpsRHtHZma3B4FHkuliwIUjZyFBftkabCtKqmbu1bRg2AU6RV6pwlleKyjDgTHf5BkD8g01coa/v+O0ClXifShEVaTrjlQY5pOy3Ctl5kj310rDJZNKr64nYTMxPT8Y5qc94UUis5FcIRdx/gReeDhvjIRxg4lzXsZYjkzsebL/haJfsKEwIK7ALiDN46w0jek2JX1AYbb0znQNuxBZb09cQDs6KpuBj8wlmameUb/ttjVDnSC9A4Yogf6MJfbnpi0zyrjK/MaOhU9ZPBW8fuIYxb2B2Usa1bOetH+IxITN1Y8dX9e87c7dqNhg7EmUY9vRH7CeuvRj7wf7W3KXBxVdCv1NMqbcabv3fnyIZxoER204aX9vk0d8ygccCbzq+xPES3Q9oa9GiF1yZPBy5k2hj5zJAWqb+FRuw3yzCNHQFYq4XhxCedg+Km+4wNuDwtQd7EU8lBNhR4ocgT2HIGZPpo/BOnmy+A6dEyhI=)

Good point, cheers for the hint.

**Comment by kubajmarek** [comment]
This causes some issues for me, as I throw an error on one getter in case some properties aren't yet set. Now most `storeToRefs` on that store throw that error, even though the getter remains unused.

Let me know if you think this is an anti-pattern.

**Comment by posva** [comment]
Yes, that seems like an anti pattern: a computed shouldn’t throw

**Comment by ezio-melotti** [comment]
I'm seeing the same issue with getters too.  Now the getters seem to be evaluated as soon as I import the store, and that causes errors because of variables in the getters that are not yet initialized.

**Comment by posva** [comment]
> Now the getters seem to be evaluated as soon as I import the store

@ezio-melotti I suppose you mean _when using `storeToRefs()`_. getters are computed properties, so this should be the same issue. Usually you catch this kind of errors with types. You can temporarily enable `// @ts-check` at top of the file to fix these issues and then remove the comment

**Comment by ezio-melotti** [comment]
@posva, thanks for your reply and for looking into this!
The fix you applied in 67d3109 should be included in v2.2.8, but even with that version I'm [getting the same error](https://github.com/overthesun/simoc-web/pull/878#issuecomment-2508827696).  Is this expected?

I understand that this change in behavior might have simply revealed already-existing errors in my code -- which is great! -- but it's nonetheless an unexpected change, especially for a bug-fix release.  I'm also somewhat concerned about how the change might affect other computed/getters that are now evaluated earlier without failures, but might have side-effect that will cause issue later on.

**Comment by posva** [comment]
computed (and getters) should not have any side effects, that’s what watchers are for
I can’t check your personal project but you can either [book a cal](https://cal.com/posva/consultancy) or provide a boiled down repro
